server/src/main/java/io/crate/expression/operator/any/AnyNeqOperator.java

```diff
@@ -108,27 +108,26 @@ protected Query literalMatchesAnyArrayRef(Function any, Literal<?> probe, Refere
         Object value = probe.value();
         BooleanQuery.Builder query = new BooleanQuery.Builder();
         query.setMinimumNumberShouldMatch(1);
-        query.add(
-            eqQuery.rangeQuery(
-                columnName, value,
-                null,
-                false,
-                false,
-                candidates.hasDocValues(),
-                candidates.indexType() != IndexType.NONE),
-            BooleanClause.Occur.SHOULD
-        );
-        query.add(
-            eqQuery.rangeQuery(
-                columnName,
-                null,
-                value,
-                false,
-                false,
-                candidates.hasDocValues(),
-                candidates.indexType() != IndexType.NONE),
-            BooleanClause.Occur.SHOULD
-        );
+        var gt = eqQuery.rangeQuery(
+            columnName, value,
+            null,
+            false,
+            false,
+            candidates.hasDocValues(),
+            candidates.indexType() != IndexType.NONE);
+        var lt = eqQuery.rangeQuery(
+            columnName,
+            null,
+            value,
+            false,
+            false,
+            candidates.hasDocValues(),
+            candidates.indexType() != IndexType.NONE);
+        if (lt == null || gt == null) {
```

Comment:
I guess that both should be null if we cannot create a rangeQuery for lt, than we cannot create also for gt, right? So maybe use `&&` and add an `assert` to verify that if one is null both are null?

Reply:
Thank you! updated